text/0033-logging.md

```diff
@@ -0,0 +1,203 @@
+```
+
+It also gives the stack trace in case of exception.
+
+#### ElasticSearch
+
+Elasticsearch is probably the most versatile search engine when it comes to logging. [See the documentation](https://www.elastic.co/guide/en/elasticsearch/reference/current/logging.html).
+
+It is possible to define a precise rolling log strategy.
+
+A rolling log strategy offers permits to:
+
+- Facilitate the administration of systems that generate a large number of logs.
+- It automates swapping, compressing, deleting, and sending logs. It assists in keeping the logging system within the specified file system space limits.
+- Defines an interval over which log analysis can be performed
+- Gives an efficient way to identify log files that are no longer used so that an automated process can clean up and compress the log directory and run log analysis programs.
+
+[See more information here about log rotation strategy](https://en.wikipedia.org/wiki/Log_rotation)
+
+This is an important thing to have when systems generate a lot of logs or are under heavy load. It's also a good practice to populate a monitoring dashboard at certain times and, through that, provide an easy way to analyze and understand what happened in the system in a human-readable way.
+
+Internally, Elasticsearch uses `log4j2` to track events and configure the logging policy. Logging levels can be configured on a per-package basis, giving the precision needed to monitor a specific feature or time in a search engine's lifecycle.
+
+A log level can be: `debug`, `info`, `warn`, `error`, `fatal`, or `unknown`.
+
+#### Elastic Entreprise Search
+
+Elasticsearch's SaaS platform, like Algolia, offers a fairly similar endpoint for browsing system-generated logs.
+
+However, it is possible to disable log storage, change the retention period and also filter the logs with more details.
+
+![image](https://www.elastic.co/guide/en/app-search/current/images/guides/log-settings-controls.png)
+*Elastic App Search's Log Retention settings view.*
+
+> It is also possible to do that from a Log Settings API endpoint.
+
+> API Logs track requests and responses at the engine level, while Analytics Logs track requests, clicks and counts.
+
+### IV. Explanation
+
+#### Current Logging behaviour of MeiliSearch (0.20)
+
+MeiliSearch uses `env_logger` to allow to define the log output level from the `RUST_LOG` environment variable. `env_logger` allows to define a specific log level per module if necessary. [See more here](https://docs.rs/env_logger/0.8.3/env_logger/).
+
+#### Logging behaviour for Milli (0.21)
+
+We have decided to keep the use of `env_logger` for Milli/Transplant. However, we will make some changes to make the logging more consistent and versatile.
+
+We will expose a `MEILI_LOG_LEVEL` environment variable that corresponds to `RUST_LOG` in the binary to improve its meaning and clarity to users. A CLI option `--log-level` will also be added to follow our configuration convention.
```

Comment:
```suggestion
#### Current Logging behaviour of MeiliSearch (0.20)

MeiliSearch uses the rust lib `env_logger` to output logs based on the log level fetched from the `RUST_LOG` environment variable. `env_logger` allows to define a specific log level per module if necessary. [See more here](https://docs.rs/env_logger/0.8.3/env_logger/).

#### Logging behaviour for Milli (0.21)

We have decided to keep using the rust lib `env_logger` for Milli/Transplant. However, we will make some changes to make the logging more consistent and versatile.

Instead of using `RUST_LOG`, we create a `MEILI_LOG_LEVEL` environment variable that corresponds to `RUST_LOG` in the binary to improve its meaning and clarity to users. A CLI option `--log-level` will also be added to follow our configuration convention.
```